cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/FieldLineageAdmin.java

```diff
@@ -158,23 +140,87 @@ FieldLineageSummary getSummary(Constants.FieldLineage.Direction direction, EndPo
     return new FieldLineageSummary(incoming, outgoing);
   }
 
-  private Set<DatasetField> convertSummaryToDatasetField(Set<EndPointField> summary) {
-    Map<EndPoint, Set<String>> endPointFields = new HashMap<>();
-    for (EndPointField endPointField : summary) {
-      EndPoint endPoint = endPointField.getEndPoint();
-      Set<String> fields = endPointFields.computeIfAbsent(endPoint, k -> new HashSet<>());
-      fields.add(endPointField.getField());
+  /**
+   * Get the summary for the specified dataset over a given time range depending on the direction specified.
+   * The summary will contain all the field level lineage relations about all the fields in a dataset.
+   *
+   * @param direction the direction in which summary need to be computed
+   * @param endPoint the EndPoint whicn represents the dataset that field level lineage needs to get computed
+   * @param start start time (inclusive) in milliseconds
+   * @param end end time (exclusive) in milliseconds
+   * @return the summary which contains all the field level lineage information about all the fields in a dataset
+   * @throws IOException if fails to get teh schema of the dataset
+   */
+  DatasetFieldLineageSummary getDatasetFieldLineage(Constants.FieldLineage.Direction direction, EndPoint endPoint,
+                                                    long start, long end) throws IOException {
+    Set<String> lineageFields = fieldLineageReader.getFields(endPoint, start, end);
+    Map<DatasetId, Set<FieldRelation>> incomingRelations = new HashMap<>();
+    Map<DatasetId, Set<FieldRelation>> outgoingRelations = new HashMap<>();
+    for (String field : lineageFields) {
+      EndPointField endPointField = new EndPointField(endPoint, field);
+
+      // compute the incoming field level lineage
+      if (direction == Constants.FieldLineage.Direction.INCOMING ||
```

Comment:
This and the next `if` block seems highly duplicated, can you refactor into a private method?